src/crawlee/_autoscaling/snapshotter.py

```diff
@@ -145,11 +159,16 @@ async def __aexit__(
         This method stops capturing snapshots of system resources (CPU, memory, event loop, and client information).
         It should be called to terminate resource capturing when it is no longer needed.
         """
-        self._event_manager.off(event=Event.SYSTEM_INFO, listener=self._snapshot_cpu)
-        self._event_manager.off(event=Event.SYSTEM_INFO, listener=self._snapshot_memory)
-        await self._snapshot_event_loop_task.stop()
-        await self._snapshot_client_task.stop()
+        if self._active:
+            self._event_manager.off(event=Event.SYSTEM_INFO, listener=self._snapshot_cpu)
+            self._event_manager.off(event=Event.SYSTEM_INFO, listener=self._snapshot_memory)
+            await self._snapshot_event_loop_task.stop()
+            await self._snapshot_client_task.stop()
+            self._active = False
+        else:
+            logger.warning(f'The {self.__class__.__name__} is not active.')
```

Comment:
I think this should raise an Exception. Once someone gets here, then they did something wrong.